packages/labs/router/src/routes.ts

```diff
@@ -163,14 +167,24 @@ export class Routes implements ReactiveController {
       // Simulate a tail group with the whole pathname
       this._currentParams = {0: tailGroup};
     } else {
-      const route = (this._currentRoute = this._getRoute(pathname));
+      const route = this._getRoute(pathname);
       if (route === undefined) {
         throw new Error(`No route found for ${pathname}`);
       }
       const pattern = getPattern(route);
-      const r = pattern.exec({pathname});
-      this._currentParams = r?.pathname.groups ?? {};
-      tailGroup = getTailGroup(this._currentParams);
+      const result = pattern.exec({pathname});
+      const params = result?.pathname.groups ?? {};
+      tailGroup = getTailGroup(params);
+      if (typeof route.enter === 'function') {
+        const success = await route.enter(params);
```

Comment:
what happens if another navigation begins before this one is finished (e.g. the user clicks a link, nothing happens for a bit, so they click it a few more times)?

Reply:
There's a TODO at the top of `goto()` related to this too.

We need to consider a few use cases here - it'd be good to gather realistic ones from the community. I can imagine cases where you want a new navigation to cancel a pending one, and some cases where you want a pending navigation to block new navigations (some kinds of navigating form submission). And forms may bring up a case where you want to block navigations from arbitrary code outside of the router. And maybe all of that _shouldn't_ be in the router because the browser doesn't disable links and what you want is a user-affordance like a modal with a scrim that disables all the visible DOM.

I'll add something to the design doc.